g4f/Provider/base_provider.py

```diff
@@ -60,26 +61,22 @@ class AsyncGeneratorProvider(AsyncProvider):
     supports_stream = True
 
     @classmethod
-    def create_completion(
+    async def create_completion(
         cls,
         model: str,
         messages: list[dict[str, str]],
         stream: bool = True,
         **kwargs
     ) -> CreateResult:
-        loop = asyncio.new_event_loop()
-        try:
-            asyncio.set_event_loop(loop)
-            generator = cls.create_async_generator(model, messages, stream=stream, **kwargs)
-            gen  = generator.__aiter__()
-            while True:
-                try:
-                    yield loop.run_until_complete(gen.__anext__())
-                except StopAsyncIteration:
-                    break
-        finally:
-            asyncio.set_event_loop(None)
-            loop.close()
+
+        while True:
+            generator  = cls.create_async_generator(model, messages, stream=stream, **kwargs)
```

Comment:
You should call create_async_generator once

Reply:
Thank you for pointing that out, I missed it, should be fixed now.